manifests/init.pp

```diff
@@ -29,8 +29,10 @@
   $tomcat = $gutterball::params::tomcat,
 ) inherits gutterball::params {
 
+  $_ = cache_data('gutterball_db_password', $dbpassword)
+  $__ = cache_data('gutterball_db_user', $dbuser)
+  $___ = cache_data('gutterball_tomcat', $tomcat)
```

Comment:
Why wouldn't Kafo have access to those top level parameters? This looks like the exact same pattern as https://github.com/theforeman/foreman-installer/blob/6d5c89088849c3bb2993df7a89cc5a84b5a95f2d/hooks/post/10-post_install_message.rb#L8-L12 where we print the (randomly generated and cached) admin password. Is there a difference between pre and post here?

In the foreman we chose to do the migrations in puppet code so users without the installer could also benefit from correct upgrades, so you could consider that as well.